client/src/components/Upload/UploadModalContent.vue

```diff
@@ -167,12 +168,50 @@ export default {
         currentFtp: function () {
             return this.currentUserId && this.ftpUploadSite;
         },
-
+        toData: function (items, history_id) {
+            var data = {
+                fetchRequest: null,
+                uploadRequest: null,
+            };
+            if (items && items.length > 0) {
+                var split = this.preprocess(items);
+                if (split.urls.length > 0) {
+                    data.fetchRequest = this.toFetchData(split.urls, history_id);
+                }
+                if (split.files.length > 0) {
+                    data.uploadRequest = this.toFileUploadData(split.files, history_id);
+                }
+            }
+            return data;
+        },
+        preprocess: function (items) {
+            var data = {
+                urls: [],
+                files: [],
+            };
+            for (var index in items) {
+                var it = items[index];
+                if (it.get("file_mode") != "new" || !this.itemIsURL(it)) {
+                    data.files.push(it);
+                } else {
+                    data.urls.push(it);
+                }
+            }
+            return data;
+        },
+        itemIsURL: function (item) {
+            for (var index in this.URI_PREFIXES) {
```

Comment:
I'm pretty sure we don't want to use var here and we could simplify the loop by using ``of`` instead of ``in``.

```javascript
for(const prefix of this.URI_PREFIXES) {
```

Reply:
I like this over explicit loops as an alternative.
```javascript
return this.URI_PREFIXES.some((prefix) => item.get('url_paste').startswith(prefix))
```